ksqldb-benchmark/src/main/java/io/confluent/ksql/benchmark/SerdeBenchmark.java

```diff
@@ -79,100 +82,166 @@
   private static final String SCHEMA_FILE_SUFFIX = ".avro";
   private static final String TOPIC_NAME = "serde_benchmark";
 
-  @State(Scope.Thread)
-  public static class SchemaAndGenericRowState {
-    LogicalSchema schema;
-    GenericRow row;
+  private static final String IMPRESSIONS_SCHEMA = "impressions";
+  private static final String METRICS_SCHEMA = "metrics";
+  private static final String SINGLE_KEY_SCHEMA = "single-key";
 
-    @Param({"impressions", "metrics"})
-    public String schemaName;
+  private static final String SEPARATOR = "/";
 
-    @Setup(Level.Iteration)
-    public void setUp() throws Exception {
-      final Generator generator = new Generator(getSchemaStream(), new Random());
+  private static final String JSON_FORMAT = "JSON";
+  private static final String AVRO_FORMAT = "Avro";
+  private static final String PROTOBUF_FORMAT = "Protobuf";
+  private static final String DELIMITED_FORMAT = "Delimited";
+  private static final String KAFKA_FORMAT = "Kafka";
 
-      // choose arbitrary key
-      final String key = generator.schema().getFields().get(0).name();
+  private static final class Params {
 
-      final RowGenerator rowGenerator = new RowGenerator(generator, key, Optional.empty());
+    final String schemaName;
+    final String formatName;
+
+    static Params parse(final String text) {
+      final String[] parts = text.split(SEPARATOR);
+      if (parts.length != 2) {
+        throw new IllegalArgumentException("Param should be in form "
+            + "'<format-name>|<schema-name>', got: " + text);
```

Comment:
The separator is `/` and you have `|`.  Also, shouldn't be `<schema-name/<format-name>` ?

Reply:
Good catch. Thanks!